src/imputation/MoR.py

```diff
@@ -234,11 +236,11 @@ def calculate_growth_rates(current_df, prev_df, target_vars):
 
     # Calculate the ratios for the relevant variables
     for target in target_vars:
-        mask = (gr_df[f"{target}_prev"] != 0) & (gr_df[target] != 0)
-        gr_df.loc[mask, f"{target}_gr"] = (
-            gr_df.loc[mask, target] / gr_df.loc[mask, f"{target}_prev"]
+        # Calculate a growth rate if both the current and previous values are non-zero
+        valid_mask = (gr_df[f"{target}_prev"] != 0) & (gr_df[target] != 0)
+        gr_df.loc[valid_mask, f"{target}_gr"] = (
```

Comment:
I am not sure what if one of them IS zero. Is the link factor set to 1, None or..